### PR TITLE
Fix quality analyzer overmatches and exceptions

### LINKED ISSUE #474: Recent changes to DB logic in server and plugins have broken the Quality Analyzer plugin
## Describe the bug
Since recent changes to DB logic in the FASTEN server and plugins, the Quality Analyzer plugin has been broken. It runs into DataAccess exceptions when accessing the DB.

These are the relevant changes:
image

## To Reproduce
1. Check out DC branch https://github.com/fasten-project/fasten-docker-deployment/tree/fasten-pipeline-release-0.0.9
2. Delete `docker-volumes/fasten`
3. `docker-compose --profile java up -d --build`
4. Browse to http://localhost:9080/api/mvn/packages/log4j:log4j/1.2.17
5. Observe the container log of fasten-docker-deployment-fasten-rapid-metadata-plugin-1 to see it run into DataAccess exceptions.

### PR DESCRIPTION
**Pull request by MagielBruntink**
## Description
Improved the matching of quality analysis results to callables in the metadata DB. In particular, when constructors contained nested classes, callables could be accidentally matched. This has been improved by requiring the uri of callables to include leading class name + the callable name itself.

## Testing
Tested locally with DC for Java, Python, and C.

Fixes #474.

### CODE REVIEW

File: analyzer/quality-analyzer/src/main/java/eu/fasten/analyzer/qualityanalyzer/QualityAnalyzerPlugin.java [hunk 1]
```diff
@@ -202,8 +186,7 @@ public void setPluginError(Exception throwable) {
 
         @Override
         public long getMaxConsumeTimeout() {
```
Comment (mir-am) [inline analyzer/quality-analyzer/src/main/java/eu/fasten/analyzer/qualityanalyzer/QualityAnalyzerPlugin.java:188]: Why is the consume timeout decreased? I assume QualityAnalyzer is faster now and doesn't need a 1h timeout value.

File: analyzer/quality-analyzer/src/test/java/eu/fasten/analyzer/qualityanalyzer/MetadataUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        assertEquals("LoggingReceiver$Slurper.%3Cinit%3E",
+                normalizeCallableName("LoggingReceiver::Slurper::Slurper"));
+
+        assertEquals("PatternParser$NamedPatternConverter.%3Cinit%3E",
+                normalizeCallableName("PatternParser::NamedPatternConverter::NamedPatternConverter"));
+    }
+
+    @Test
+    void normalizeCallableNamePythonTest() {
+        assertEquals("get_filing_list",
```
Comment (mir-am) [inline analyzer/quality-analyzer/src/test/java/eu/fasten/analyzer/qualityanalyzer/MetadataUtilsTest.java:33]: I do not see a difference between expected output and generated output by `normalizeCallableName`. I guess `normalizeCallableName` only works for Java callables?

**Review by mir-am — APPROVED** [review]
Thanks for the PR.
Overall, LGTM. I've made some minor comments.

File: analyzer/quality-analyzer/src/test/java/eu/fasten/analyzer/qualityanalyzer/MetadataUtilsTest.java (hunk 2 above)
Comment (MagielBruntink) [reply analyzer/quality-analyzer/src/test/java/eu/fasten/analyzer/qualityanalyzer/MetadataUtilsTest.java:33]: That's true, I only added these Python and C cases to make sure that they are not somehow broken. Also to add more different examples later if we find them (don't think so).

**Pull request merged**